<!DOCTYPE html>
<html lang="en">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <title>
    Linux下BLAST的安装与使用 |
    
    Huan&#39;s blog</title>
  
    <link rel="shortcut icon" href="/favicon.ico">
  
  <link rel="stylesheet" href="/css/style.css">
  
    <link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">
  
  <script src="/js/pace.min.js"></script>
</head>
</html>
<body>
<main class="content">
  <section class="outer">
  

<article id="post-blast" class="article article-type-post" itemscope itemprop="blogPost" data-scroll-reveal>
  
  <div class="article-inner">
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      Linux下BLAST的安装与使用
    </h1>
  
  




      </header>
    

    
      <div class="article-meta">
        <a href="/2017/12/16/blast/" class="article-date">
  <time datetime="2017-12-16T12:54:31.000Z" itemprop="datePublished">2017-12-16</time>
</a>
        
      </div>
    

    
      




    

    <div class="article-entry" itemprop="articleBody">
      


      

      
        <p>原本计划对几个亚洲棉基因进行blast比对寻找在陆地棉中的同源基因，但是服务器抽风了，导致计划被打乱，不过刚好也乘此机会学习和总结一下BLAST+的安装和使用。<br>本文实测环境为Linux64位系统。<a id="more"></a></p>
<p>##1.安装配置BLAST+<br>首先是下载程序，</p>
<pre><code>wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.2.30+-x64-linux.tar.gz
</code></pre><p>对文件解压</p>
<pre><code>tar -zxvf ncbi-blast-2.2.30+-x64-linux.tar.gz
</code></pre><p>为了方便管理，更喜欢将其移至我的本地安装目录中</p>
<pre><code>mv ncbi-blast-2.2.30+ ~/local/app/ # 移动
cd ~/local/app/                    # 进入本地程序安装路径
mv ncbi-blast-2.2.30+ blast        # 修改目录名
</code></pre><p>这样就安装到我平时安装本地软件的文件夹里了，当然和其他软件的安装方法一样，你需要将blast+的路径加入到环境变量中，方便以后的调取。</p>
<pre><code>echo &quot;export PATH=/db/home/shenwei/local/app/blast/bin:\$PATH&quot; &gt;&gt; ~/.bashrc
source ~/.bashrc
</code></pre><p>现在一切安装工作已经就绪了</p>
<p>##2.配置本地的BLAST库<br>为了方便大批量的比对工作，提高工作效率，本地化库是个很好的方法。<br>建立并编辑BLAST全局配置文件</p>
<pre><code>[hsi@node102 ~]$ cat .ncbirc  #内容如下

[BLAST]
BLASTDB=/home/hsi/blast/db
</code></pre><p>这样配置的好处在于以后调用库的时候不需要写绝对路径，只输入名称就可以。</p>
<p>##3.准备库文件<br>因为我本地文件夹中有一套fastq格式的基因组序列，所以我直接就用本地文件，不再去下载了，这一部分就不再包括如何下载，如何更新的内容了。</p>
<p>FAST格式的文件不能作为库直接用于本地的blast，必须要利用makeblastdb进行格式化</p>
<pre><code>makeblastdb -in input_db -dbtype nucl -parse_seqids -out dbname
</code></pre><p>-in: 后加输入文件<br>-dbtype: 代表数据库的类型，nucl是核酸类型，prot是蛋白质类型</p>
<p>如果该库需要经常使用，可将库文件移到前面配置的库文件的目录，今后在其它目录运行blast的时候，便可直接输入库名（不用输入绝对路径），直接使用。</p>
<pre><code>mv dbname.* ~/data/blast
</code></pre><p>##4.执行BLAST</p>
<pre><code>blastn -query test.fa -db nt -outfmt 6 -evalue 1e-5 -out &quot;test.blastn@dbname&quot; -num_threads 8
</code></pre><p>参数说明:<br>-query： 输入文件路径及文件名<br>-out：输出文件路径及文件名<br>-db：格式化了的数据库路径及数据库名<br>-outfmt：输出文件格式，总共有12种格式,6对应的是blast的m8格式<br>-evalue：设置输出结果的e-value值<br>-num_descriptions：tabular格式输出结果的条数<br>-num_threads：线程数</p>
<p>结果文件解读：<br>进行Blast+比对，用参数-outfmt 6可以以列表的方式输出结果，结果中从左到右每一列的意义分别是：</p>
<p>[00] Query id</p>
<p>[01] Subject id</p>
<p>[02] % identity</p>
<p>[03] alignment length</p>
<p>[04] mismatches</p>
<p>[05] gap openings</p>
<p>[06] q. start</p>
<p>[07] q. end</p>
<p>[08] s. start</p>
<p>[09] s. end</p>
<p>[10] e-value</p>
<p>[11] bit score</p>
<p>核酸序列比对核酸数据库（blastn）以及核酸序列比对蛋白数据库（blastx）<br>与上面的blastp用法类似：</p>
<pre><code>blastn -query seq.fasta -out seq.blast -db dbname -outfmt 6 -evalue 1e-5 -num_descriptions 10 -num_threads 8

blastx -query seq.fasta -out seq.blast -db dbname -outfmt 6 -evalue 1e-5 -num_descriptions 10 -num_threads 8
</code></pre><p>详情请参考<a href="https://www.ncbi.nlm.nih.gov/books/NBK279690/" target="_blank" rel="noopener">《BLAST® Command Line Applications User Manual》</a>，<a href="http://www.yelinsky.com/blog/archives/421.html" target="_blank" rel="noopener">BLAST+使用方法</a>，<a href="http://blog.shenwei.me/local-blast-installation/" target="_blank" rel="noopener">《Linux系统中NCBI BLAST+本地化教程》</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://huans.github.io/2017/12/16/blast/" data-id="ck2nb4e4f0006k495nxce94jd"
         class="article-share-link">Share</a>
      
    </footer>

  </div>

  
    
  <nav class="article-nav">
    
      <a href="/2017/12/18/QTL/" class="article-nav-link">
        <strong class="article-nav-caption">Newer posts</strong>
        <div class="article-nav-title">
          
            连锁分析与关联分析的区别
          
        </div>
      </a>
    
    
      <a href="/2017/12/14/Disqus使用及报错处理/" class="article-nav-link">
        <strong class="article-nav-caption">Olde posts</strong>
        <div class="article-nav-title">Disqus使用及报错处理</div>
      </a>
    
  </nav>


  

  
    
  <div class="gitalk" id="gitalk-container"></div>
  <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
  <script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>
  <script src="https://cdn.bootcss.com/blueimp-md5/2.10.0/js/md5.min.js"></script>
  <script type="text/javascript">
    var gitalk = new Gitalk({
      clientID: '364195da21f40dada57e',
      clientSecret: 'ce80c7601ca7a72a5f17d1f4eb860ea9ef86d57a',
      repo: 'blog-comments',
      owner: 'huans',
      admin: ['huans'],
      // id: location.pathname,      // Ensure uniqueness and length less than 50
      id: md5(location.pathname),
      distractionFreeMode: false,  // Facebook-like distraction free mode
      pagerDirection: 'last'
    })

  gitalk.render('gitalk-container')
  </script>

  

</article>



</section>
  <footer class="footer">
  <div class="outer">
    <div class="float-right">
      <ul class="list-inline">
  
    <li><i class="fe fe-smile-alt"></i> <span id="busuanzi_value_site_uv"></span></li>
  
</ul>
    </div>
    <ul class="list-inline">
      <li>&copy; 2019 Huan&#39;s blog</li>
      <li>Powered by <a href="http://hexo.io/" target="_blank">Hexo</a></li>
      <li>Theme  <a href="https://github.com/zhwangart/hexo-theme-ocean">Ocean</a></li>
    </ul>
  </div>
</footer>

</main>

<aside class="sidebar sidebar-specter">
  
    <button class="navbar-toggle"></button>
<nav class="navbar">
  
    <div class="logo">
      <a href="/"><img src="/images/hexo.svg" alt="Huan&#39;s blog"></a>
    </div>
  
  <ul class="nav nav-main">
    
      <li class="nav-item">
        <a class="nav-item-link" href="/">Home</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/archives">Archives</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/gallery">Gallery</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/about">About</a>
      </li>
    
    <li class="nav-item">
      <a class="nav-item-link nav-item-search" title="Search">
        <i class="fe fe-search"></i>
        Search
      </a>
    </li>
  </ul>
</nav>
<nav class="navbar navbar-bottom">
  <ul class="nav">
    <li class="nav-item">
      <div class="totop" id="totop">
  <i class="fe fe-rocket"></i>
</div>
    </li>
    <li class="nav-item">
      
        <a class="nav-item-link" target="_blank" href="/atom.xml" title="RSS Feed">
          <i class="fe fe-feed"></i>
        </a>
      
    </li>
  </ul>
</nav>
<div class="search-form-wrap">
  <div class="local-search local-search-plugin">
  <input type="search" id="local-search-input" class="local-search-input" placeholder="Search...">
  <div id="local-search-result" class="local-search-result"></div>
</div>
</div>
  </aside>
  <script src="/js/jquery-2.0.3.min.js"></script>
<script src="/js/jquery.justifiedGallery.min.js"></script>
<script src="/js/lazyload.min.js"></script>
<script src="/js/busuanzi-2.3.pure.min.js"></script>

  <script src="/fancybox/jquery.fancybox.min.js"></script>



  <script src="/js/tocbot.min.js"></script>
  <script>
    // Tocbot_v4.7.0  http://tscanlin.github.io/tocbot/
    tocbot.init({
      tocSelector: '.tocbot',
      contentSelector: '.article-entry',
      headingSelector: 'h1, h2, h3, h4, h5, h6',
      hasInnerContainers: true,
      scrollSmooth: true,
      positionFixedSelector: '.tocbot',
      positionFixedClass: 'is-position-fixed',
      fixedSidebarOffset: 'auto',
    });
  </script>


<script src="/js/ocean.js"></script>

</body>
</html>