<!DOCTYPE html>
<html lang="en">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="utf-8">
  
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <title>
    sentieon——更具效率的变异检测软件 |
    
    Huan&#39;s blog</title>
  
    <link rel="shortcut icon" href="/favicon.ico">
  
  <link rel="stylesheet" href="/css/style.css">
  
    <link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">
  
  <script src="/js/pace.min.js"></script>
</head>
</html>
<body>
<main class="content">
  <section class="outer">
  

<article id="post-sentieon" class="article article-type-post" itemscope itemprop="blogPost" data-scroll-reveal>
  
  <div class="article-inner">
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      sentieon——更具效率的变异检测软件
    </h1>
  
  




      </header>
    

    
      <div class="article-meta">
        <a href="/2019/11/05/sentieon/" class="article-date">
  <time datetime="2019-11-05T13:11:40.354Z" itemprop="datePublished">2019-11-05</time>
</a>
        
      </div>
    

    
      




    

    <div class="article-entry" itemprop="articleBody">
      


      

      
        <p>目前在全基因组范围内做variants calling的软件还是GATK，Samtools为主，我之前的工作基本上属于这两个都用一点，昨天需要跑一批数据的时候发现GATK更新了，很多命令和参数都有了改变，觉得再去看manual就很麻烦，反正都是看不如就学习一下最近非常火的sentieon。<a id="more"></a><br>sentieon是一款商业软件（据说价格很贵），相比于目前主流的软件，它的主要特点就是 <strong>快</strong>、<strong>准</strong>。<br>相比于GATK，sentieon用的和GATK一样的数学模型，不一样的是对算法进行了改进，用sentieon的结果和GATK一样，而速度据说能比GATK快20-50倍。<br>因为我们实验室购买了这个软件，也都全部安装配置到服务器上可以直接调用，下面就贴一下我使用过程中的脚本，十分感谢杨庆勇老师在群里的无私分享。<br>因为样本比较多，所以我是通过批量提交作业的方法进行的。</p>
<p>#batch_run.lsf</p>
<pre><code> #BSUB -J sentieon
    #BSUB -n 1
    #BSUB -R span[hosts=1]
    #BSUB -o %J.out
    #BSUB -e %J.err

for sample in ./SRR*_2.fastq.gz
        do
        sh sentieon.sh ${sample} &amp; 
        sleep 1
done
wait
</code></pre><p>#sentieon.sh</p>
<pre><code>#!/bin/sh
#样本名称
sample=$1
i=$(basename $sample |sed &apos;s/_2.fastq.gz//&apos;)
# 加载所需软件
module load sentieon/201808.07
export SENTIEON_LICENSE=mn01:9000
module load SAMtools/1.9
release_dir=/public/home/software/opt/bio/software/Sentieon/201808.07

# fastq文件路径
fq1=${i}_1.fastq.gz
fq2=${i}_2.fastq.gz
# 参考基因组文件
fasta=xxx.fasta
# 输出文件路径
workdir=`pwd`/sentieon_result
# 需要使用的核心数
nt=10
# 比对信息
group_prefix=&quot;read_group_name&quot;
platform=&quot;ILLUMINA&quot;
mq=30

[ ! -d $workdir ] &amp;&amp; mkdir -p $workdir
cd $workdir
# 输出文件
rawCram=$i.cram
sortedCram=$i.q$mq.sorted.cram
depCram=$i.deduped.cram
realnCram=$i.realn.cram
outvcf=$i.sentieon.vcf
exec &gt; $workdir/$i.callVCF.log 2&gt;&amp;1 # call vcf的日志文件
# ******************************************
# 1. 利用 BWA-MEM 进行比对并排序
# ******************************************
bsub -K -J Sentieon -n 10 -o %J.out -e %J.err -R span[hosts=1] &quot;( $release_dir/bin/sentieon bwa mem -M -R \&quot;@RG\tID:${i}\tSM:${i}\tPL:$platform\&quot; \
-t $nt -K 10000000 $fasta $fq1 $fq2 || echo -n &apos;error&apos; ) | samtools sort -@ $nt  --output-fmt CRAM \
--reference $fasta -o $rawCram - &amp;&amp; samtools index -@ $nt $rawCram 
samtools view -hCS -T $fasta -q $mq -o $sortedCram $rawCram &amp;&amp; \
samtools index -@ $nt $sortedCram
samtools flagstat $rawCram &gt; $i.stat.raw.txt &amp;&amp; \
samtools flagstat $sortedCram &gt; $i.stat.q$mq.txt &amp;

# ******************************************
# 2. Calculate data metrics
# ******************************************
$release_dir/bin/sentieon driver -r $fasta -t $nt -i $sortedCram --algo MeanQualityByCycle ${i}_mq_metrics.txt \
--algo QualDistribution ${i}_qd_metrics.txt --algo GCBias --summary ${i}_gc_summary.txt ${i}_gc_metrics.txt \
--algo AlignmentStat --adapter_seq &apos;&apos; ${i}_aln_metrics.txt --algo InsertSizeMetricAlgo ${i}_is_metrics.txt
$release_dir/bin/sentieon plot metrics -o ${i}_metrics-report.pdf gc=${i}_gc_metrics.txt \
qd=${i}_qd_metrics.txt mq=${i}_mq_metrics.txt isize=${i}_is_metrics.txt
$release_dir/bin/sentieon driver -r $fasta -t $nt -i $sortedCram --algo LocusCollector --fun score_info ${i}_score.txt

# ******************************************
# 3. 去除 Duplicate Reads
# ******************************************
$release_dir/bin/sentieon driver -r $fasta -t $nt -i $sortedCram --algo Dedup --rmdup --cram_write_options version=3.0 \
--score_info ${i}_score.txt --metrics ${i}_dedup_metrics.txt $depCram &amp;&amp; rm -f $sortedCram

# ******************************************
# 4. Indel 重排序 (可选)
# 如果只需要最终的比对结果文件，到这里就可以了，这条命令下面的命令都可以注释掉
# ******************************************
$release_dir/bin/sentieon driver -r $fasta -t $nt -i $depCram --algo Realigner --cram_write_options version=3.0 \
$realnCram &amp;&amp; rm -f $depCram

# ******************************************
# 5. Variant calling
# ******************************************
$release_dir/bin/sentieon driver -t $nt -r $fasta -i $realnCram --algo Genotyper $outvcf&quot;
</code></pre><p>需要注意的是在这个地方：</p>
<blockquote>
<p>bsub -K -J Sentieon -n 10 -o %J.out -e %J.err -R span[hosts=1] “( $release_dir/bin/sentieon bwa mem -M -R \”@RG\tID:${i}\tSM:${i}\tPL:$platform\”<br>用了”\”转义了头文件参数的两个引号，否则在执行的时候会因为提交作业命令后的引号（第一个）而引起不能识别-R后面的参数。</p>
</blockquote>

      
    </div>
    <footer class="article-footer">
      <a data-url="https://huans.github.io/2019/11/05/sentieon/" data-id="ck2lwf7940008h89594h4dw0w"
         class="article-share-link">Share</a>
      
  <ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/sentieon-GATK-variants-calling/">sentieon GATK variants-calling</a></li></ul>

    </footer>

  </div>

  
    
  <nav class="article-nav">
    
    
      <a href="/2019/11/04/XP-CLR/" class="article-nav-link">
        <strong class="article-nav-caption">Olde posts</strong>
        <div class="article-nav-title">群体遗传π值和XP-CLR计算</div>
      </a>
    
  </nav>


  

  
    
  <div class="gitalk" id="gitalk-container"></div>
  <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
  <script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>
  <script src="https://cdn.bootcss.com/blueimp-md5/2.10.0/js/md5.min.js"></script>
  <script type="text/javascript">
    var gitalk = new Gitalk({
      clientID: '364195da21f40dada57e',
      clientSecret: 'ce80c7601ca7a72a5f17d1f4eb860ea9ef86d57a',
      repo: 'blog-comments',
      owner: 'huans',
      admin: ['huans'],
      // id: location.pathname,      // Ensure uniqueness and length less than 50
      id: md5(location.pathname),
      distractionFreeMode: false,  // Facebook-like distraction free mode
      pagerDirection: 'last'
    })

  gitalk.render('gitalk-container')
  </script>

  

</article>



</section>
  <footer class="footer">
  <div class="outer">
    <div class="float-right">
      <ul class="list-inline">
  
    <li><i class="fe fe-smile-alt"></i> <span id="busuanzi_value_site_uv"></span></li>
  
</ul>
    </div>
    <ul class="list-inline">
      <li>&copy; 2019 Huan&#39;s blog</li>
      <li>Powered by <a href="http://hexo.io/" target="_blank">Hexo</a></li>
      <li>Theme  <a href="https://github.com/zhwangart/hexo-theme-ocean">Ocean</a></li>
    </ul>
  </div>
</footer>

</main>

<aside class="sidebar sidebar-specter">
  
    <button class="navbar-toggle"></button>
<nav class="navbar">
  
    <div class="logo">
      <a href="/"><img src="/images/hexo.svg" alt="Huan&#39;s blog"></a>
    </div>
  
  <ul class="nav nav-main">
    
      <li class="nav-item">
        <a class="nav-item-link" href="/">Home</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/archives">Archives</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/gallery">Gallery</a>
      </li>
    
      <li class="nav-item">
        <a class="nav-item-link" href="/about">About</a>
      </li>
    
    <li class="nav-item">
      <a class="nav-item-link nav-item-search" title="Search">
        <i class="fe fe-search"></i>
        Search
      </a>
    </li>
  </ul>
</nav>
<nav class="navbar navbar-bottom">
  <ul class="nav">
    <li class="nav-item">
      <div class="totop" id="totop">
  <i class="fe fe-rocket"></i>
</div>
    </li>
    <li class="nav-item">
      
        <a class="nav-item-link" target="_blank" href="/atom.xml" title="RSS Feed">
          <i class="fe fe-feed"></i>
        </a>
      
    </li>
  </ul>
</nav>
<div class="search-form-wrap">
  <div class="local-search local-search-plugin">
  <input type="search" id="local-search-input" class="local-search-input" placeholder="Search...">
  <div id="local-search-result" class="local-search-result"></div>
</div>
</div>
  </aside>
  <script src="/js/jquery-2.0.3.min.js"></script>
<script src="/js/jquery.justifiedGallery.min.js"></script>
<script src="/js/lazyload.min.js"></script>
<script src="/js/busuanzi-2.3.pure.min.js"></script>

  <script src="/fancybox/jquery.fancybox.min.js"></script>



  <script src="/js/tocbot.min.js"></script>
  <script>
    // Tocbot_v4.7.0  http://tscanlin.github.io/tocbot/
    tocbot.init({
      tocSelector: '.tocbot',
      contentSelector: '.article-entry',
      headingSelector: 'h1, h2, h3, h4, h5, h6',
      hasInnerContainers: true,
      scrollSmooth: true,
      positionFixedSelector: '.tocbot',
      positionFixedClass: 'is-position-fixed',
      fixedSidebarOffset: 'auto',
    });
  </script>


<script src="/js/ocean.js"></script>

</body>
</html>